### ISSUE TITLE
`pytest>=7` deprecation warnings 

### ISSUE THREAD

**Issue by bibajz** [HypothesisWorks/hypothesis#3222]
While using `pytest==7.0.0rc1`, I noticed some deprecation warnings in Hypothesis testing, namely:

```
  /home/libor/.virtualenvs/p310/lib/python3.10/site-packages/hypothesis/internal/escalation.py:131: PytestDeprecationWarning: A private pytest class or function was used.
    return str(item.repr_failure(ExceptionInfo((type(err), err, tb)))) + "\n"
```

So, after consulting the release notes - https://docs.pytest.org/en/7.0.x/changelog.html#deprecations , `ExceptionInfo` moved from `_pytest._code` to public `pytest` namespace.

Going through the whole code base and searching for relevant `_pytest` usages reveals:

```
$ grep -rHn hypothesis-python
hypothesis-python/tests/cover/test_mock.py:17:from _pytest.config import Config
hypothesis-python/src/hypothesis/internal/escalation.py:128:    if current_pytest_item.value is not None and "_pytest._code" in sys.modules:
hypothesis-python/src/hypothesis/internal/escalation.py:129:        item = current_pytest_item.value
hypothesis-python/src/hypothesis/internal/escalation.py:130:        ExceptionInfo = sys.modules["_pytest._code"].ExceptionInfo
...
hypothesis-python/src/hypothesis/core.py:483:    if "_pytest" in sys.modules:
hypothesis-python/src/hypothesis/core.py:484:        exceptions.add(sys.modules["_pytest"].outcomes.Skipped)
hypothesis-python/src/hypothesis/core.py:499:    if "_pytest" in sys.modules:
hypothesis-python/src/hypothesis/core.py:500:        exceptions.append(sys.modules["_pytest"].outcomes.Failed)
hypothesis-python/tests/cover/test_stateful.py:15:from _pytest.outcomes import Failed, Skipped
hypothesis-python/tests/cover/test_core.py:12:from _pytest.outcomes import Failed, Skipped
...
```

`Failed/Skipped` is still in `_pytest.outcomes`, but `Config` and `ExceptionInfo` moved to `pytest` namespace.

**Comment by Zac-HD** [comment]
Fixed in #3221 :tada:

**Comment by bibajz** [comment]
Nice! Good idea to extract the check of `pytest/_pytest.code` into separate function.

Thank you,
Libor
